### PR TITLE
add nodes info such as hardware, packages and plugins

### PR DESCRIPTION
**Pull request by ydayagi**
https://issues.redhat.com/browse/CNFCERT-24
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

**Comment by jun-rh** [comment]
@ydayagi Please address smoke test errors

**Comment by jun-rh** [comment]
Also, do we need to re-generate catalog? I wish it can be automated.

**Comment by jun-rh** [comment]
> Also, do we need to re-generate catalog? I wish it can be automated.

build-catalog-md?

**Comment by ydayagi** [comment]
> > Also, do we need to re-generate catalog? I wish it can be automated.
> 
> build-catalog-md?

fixed

**Review by jun-rh — DISMISSED** [review]
Looks good

File: test-network-function/diagnostic/suite.go [hunk 1]
```diff
@@ -82,10 +92,199 @@ var _ = ginkgo.Describe(testSuiteSpec, func() {
...
+type NodesHwInfo struct {
+	Master NodeHwInfo
+	Worker NodeHwInfo
+}
+
 // GetNodeSummary returns the result of running `oc get nodes -o json`.
 func GetNodeSummary() map[string]interface{} {
 	return nodeSummary
 }
+
```
[the remark this answers is not in the document]
Comment (ydayagi) [reply test-network-function/diagnostic/suite.go:131]: generic has one file as well with much more code. I, personally, don't see a problem with it. There's not much code and there's no real logical/functional split we can do.
There are always advantages and disadvantages to having one file. One advantage is the ease of finding the code that handles a certain part of the output.

[the remark this answers is not in the document]
Comment (jun-rh) [reply test-network-function/diagnostic/suite.go:131]: I agree with Salah. The hardware/plugin info related utility functions are large enough and unique enough to form a separate logical piece, especially when we know there will be more code like this.
@ydayagi Frank left a comment in the JIRA. Any idea how to collect CSI and service mesh plugin info?

[the remark this answers is not in the document]
Comment (ydayagi) [reply test-network-function/diagnostic/suite.go:131]: @jun-rh can we put the other plugins in a separate task?

[the remark this answers is not in the document]
Comment (jun-rh) [reply test-network-function/diagnostic/suite.go:131]: @ydayagi Sure.

**Review by HamadiSE — APPROVED** [review]

**Pull request merged**